### PR TITLE
perp-1232: Ability to add referer header to the outgoing request

### PR DESCRIPTION
**Pull request by lvn-hasky-dragon**
Unfortunately I haven't been able to test this (apart from the fact that the code path gets exercised).

I can probably explain how I attempted testing this: Ran a localosmosis locally, but it doesn't seem to provide any option to log request/response headers etc. It does provide an option `--log_level`, but that doesn't seem to be providing the level of detail I want. I would be happy to know if there is some other way of testing this patch.

### CODE REVIEW

File: packages/cosmos-bin/src/main.rs [hunk 1]
```diff
@@ -57,6 +57,10 @@ struct Opt {
         env = "COSMOS_GAS_MULTIPLIER"
     )]
     gas_multiplier: Option<f64>,
+
+    /// Referer header
+    #[clap(long, short, global = true)]
```
Comment (lvn-rusty-dragon) [inline packages/cosmos-bin/src/main.rs]: I think it could be convenient to have an env var for this too.

**Review by lvn-rusty-dragon — APPROVED** [review]
As far as testing, my best recommendation would be to set up a reverse proxy that logs the request headers and target that. Even hitting a fake server may be sufficient.

**Comment by lvn-hasky-dragon** [comment]
Thanks for the feedback, I will setup a reverse proxy and confirm the behavior. Thanks!

**Comment by lvn-hasky-dragon** [comment]
I can confirm that it adds the referrer header when tested via reverse proxy:

```
Request {
  host                 = "localhost"
  port                 = 9090
  secure               = False
  requestHeaders       = [("X-Real-IP","127.0.0.1"),("Host","localhost:3000"),("TE","trailers"),("Content-Type","application/grpc"),("Referer","https://goo.com"),("User-Agent","tonic/0.8.3")]
  path                 = "/cosmos.bank.v1beta1.Query/AllBalances"
  queryString          = ""
  method               = "POST"
  proxy                = Nothing
  rawBody              = False
  redirectCount        = 0
  responseTimeout      = ResponseTimeoutNone
  requestVersion       = HTTP/1.1
  proxySecureMode      = ProxySecureWithConnect
}
```

Although, reverse proxy was not able to parse the response properly - this confirms that this PR works as intended.

File: packages/cosmos-bin/src/main.rs (hunk 1 above)
Comment (lvn-hasky-dragon) [reply packages/cosmos-bin/src/main.rs]: Pushed a fix to recognize environment variable.

**Pull request merged**